File: backend/src/main/java/capstone/rt04/retailbackend/request/customer/UpdateShoppingCartRequest.java
package capstone.rt04.retailbackend.request.customer;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.NotNull;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UpdateShoppingCartRequest {

    @NotNull
    private Integer quantity;

    @NotNull
    private Long productVariantId;

    @NotNull
    private Long customerId;

    @NotNull
    private String cartType;
}

File: frontend/customer-web-app/src/models/ShoppingCart/UpdateShoppingCartRequest.js
export default class UpdateShoppingCartRequest {
  quantity;
  productVariantId;
  customerId;
  cartType;

  constructor(quantity, productVariantId, customerId, cartType) {
    this.quantity = quantity;
    this.productVariantId = productVariantId;
    this.customerId = customerId;
    this.cartType = cartType;
  }
}
